helper/resource/testing.go

```diff
@@ -398,7 +402,18 @@ func Test(t TestT, c TestCase) {
 	errored := false
 	for i, step := range c.Steps {
 		var err error
-		log.Printf("[WARN] Test: Executing step %d", i)
+		log.Printf("[DEBUG] Test: Executing step %d", i)
+
+		if step.SkipFunc != nil {
+			skip, err := step.SkipFunc()
```

Comment:
Looking at your example, I see that the test functions here end up reaching out to the global `testAccProvider` to see what mode it's running in.

Maybe that's fine, I guess... it feels a little counter-intuitive on first read. The "functional-style" proponent in me kinda wants to pass in some sort of contextual object here to work from, but looking above I don't see anything that would actually be useful here, so I guess leaning on the globals is okay.

Reply:
The global variable does feel wrong, but I don't think the `SkipFunc` is any special in this context. It would be better to pass the context (provider's `meta`) through to those helpers more elegantly somehow, but that would likely require more significant refactoring of the whole package, IMO.